guides/source/caching_with_rails.md

```diff
@@ -42,105 +50,98 @@ INFO: Action Caching has been removed from Rails 4. See the [actionpack-action_c
-  All available products:
+<% cache_if admin?, product do %>
+  <%= render product %>
 <% end %>
 ```
 
-This fragment is then available to all actions in the `ProductsController` using the key and can be expired the same way:
+### Russian Doll Caching
 
-```ruby
-expire_fragment('all_available_products')
-```
-If you want to avoid expiring the fragment manually, whenever an action updates a product, you can define a helper method:
-
-```ruby
-module ProductsHelper
-  def cache_key_for_products
-    count          = Product.count
-    max_updated_at = Product.maximum(:updated_at).try(:utc).try(:to_s, :number)
-    "products/all-#{count}-#{max_updated_at}"
-  end
-end
-```
+You may want to nest cached fragments inside other cached fragments. This is
+called Russian doll caching.
 
-This method generates a cache key that depends on all products and can be used in the view:
+The advantage of Russian doll caching is that if a single product is updated, all the other inner fragments can be reused when regenerating the outer fragment.
 
-```erb
-<% cache(cache_key_for_products) do %>
-  All available products:
-<% end %>
-```
+As explained in the previous section, a cached file will expire if the value of
+`updated_at` changes in a record on which the cached file directly depends.
+However, this will not expire any cache the fragment is nested within.
 
-If you want to cache a fragment under certain conditions, you can use `cache_if` or `cache_unless`
+For example, take the following view:
 
 ```erb
-<% cache_if (condition, cache_key_for_products) do %>
-  All available products:
+<% cache product do %>
+  <%= render product.games %>
 <% end %>
 ```
 
-You can also use an Active Record model as the cache key:
+Which in turn renders this view:
 
 ```erb
-<% Product.all.each do |p| %>
-  <% cache(p) do %>
-    <%= link_to p.name, product_url(p) %>
-  <% end %>
+<% cache game %>
+  <%= render game %>
 <% end %>
 ```
 
-Behind the scenes, a method called `cache_key` will be invoked on the model and it returns a string like `products/23-20130109142513`. The cache key includes the model name, the id and finally the updated_at timestamp. Thus it will automatically generate a new fragment when the product is updated because the key changes.
+If any attribute of game is changed, the `updated_at` value will be set to the
+current `DateTime`, thereby expiring the cache. However, because `updated_at`
+will not be changed for the product object that cache will not be expired and
+your app will serve stale data. To fix this, we tie the models together with
+the `touch` method:
 
-You can also combine the two schemes which is called "Russian Doll Caching":
-
-```erb
-<% cache(cache_key_for_products) do %>
-  All available products:
-  <% Product.all.each do |p| %>
-    <% cache(p) do %>
-      <%= link_to p.name, product_url(p) %>
-    <% end %>
-  <% end %>
-<% end %>
+```ruby
+class Product < ActiveRecord::Base
+end
+class Game < ActiveRecord::Base
```

Comment:
There's a missing `has_many` statement for the `Product` model here, no ? Also, could you add a blank line between the two class definitions. ?